runtime/lua/vim/lsp.lua

```diff
@@ -1057,11 +1057,38 @@ function lsp._vim_exit_handler()
     client.stop()
   end
 
-  if not vim.wait(500, function() return tbl_isempty(active_clients) end, 50) then
-    for _, client in pairs(active_clients) do
-      client.stop(true)
-    end
+  local function wait_async(timeout, ms, f, cb)
+    local timer = uv.new_timer()
+    local time = 0
+
+    timer:start(0, ms, function()
+      local function done(in_time)
+        timer:stop()
+        timer:close()
+        cb(in_time)
+      end
```

Comment:
I think this function can be defined outside of the function -- I think it's a bit confusing seeing it here.

Reply:
(like just two lines above)